pkg/recorder/recorder.go

```diff
@@ -0,0 +1,78 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package recorder
+
+import (
+	"fmt"
+
+	"github.com/golang/glog"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/kubernetes"
+	typedcorev1 "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/client-go/tools/record"
+)
+
+// Provider knows how to generate new event recorders with given name.
+type Provider interface {
+	// NewRecorder returns an EventRecorder with given name.
+	GetEventRecorderFor(name string) (record.EventRecorder, error)
+	// SetClientSet specifies the Clientset of the provider.
+	SetClientSet(cs *kubernetes.Clientset)
+	// SetScheme specifies the Scheme of the provider.
+	SetScheme(scheme *runtime.Scheme)
+}
+
+type provider struct {
+	clientSet        *kubernetes.Clientset
+	scheme           *runtime.Scheme
+	eventBroadcaster record.EventBroadcaster
+}
+
+// NewProvider create a new Provider instance.
+func NewProvider() Provider {
```

Comment:
Lets move  `New` under `pkg/internal` until we are sure we know users will need to get informers directly.  They should be able to get them through the manager.Manager.